### ISSUE TITLE
Open & Close Window Within Statement Scope Not Updating Application.Current.Windows Property

### ISSUE THREAD

**Issue by cgp1976** [dotnet/maui#18037]
### Description

I noticed that when OpenWindow and CloseWindow are within the same scope, they are causing the application to report wrong number of window instances in Application.Current.Windows and hence memory stay forever for those closed window

```cs
private void Open_Window(object sender, EventArgs e)
{

    var page = new ChildWindow();

    Application.Current.OpenWindow(new Window(page));

    var currentWindow = (sender as View).GetVisualElementWindow() as Window;

    Application.Current.CloseWindow(currentWindow);

    var count = Application.Current.Windows.Count();

    this.OnPropertyChanged(nameof(Count));
}
```

[MauiApp333.zip](https://github.com/dotnet/maui/files/12911951/MauiApp333.zip)

### Steps to Reproduce

1. Click on Open Window on (Window A)
2. On the newly Opened Window (Window B), Click on Check Window Count button. it's reporting 2 instead of 1.
3. To do further testing, you can use those available buttons to open/close/check count/Invoke GC
4. From here onwards, Invoke GC Collect Button,
5. Go Diagnostic Tools > Memory Usage and Check for the ChildWindow Class. all closed windows are still there.

### Version with bug

8.0.0-rc.2.9373

### Is this a regression from previous behavior?

No, this is something new

### Last version that worked well

Unknown/Other

### Affected platforms

Windows

### Affected platform versions

Window 11 Arm

### Did you find any workaround?

nil

### Relevant log output

```shell
nil
```

**Comment by drasticactions** [comment]
For your first issue:

https://github.com/dotnet/maui/blob/cd9fdd75510cecdd298c80eaf18fff47a5e89d7f/src/Controls/src/Core/Application/Application.cs#L455

The window is not being removed because of the Default MAUI template and how it sets up Windows. It sets "MainPage" in the initial App session. That triggers this code above so when you try to remove the window, it goes "This is the MainPage of the app, don't remove the window!" so the instance of the Window goes, but the reference hangs in the list. You can fix that by overriding CreateWindow in your app:

```csharp
    public partial class App : Application
    {
        public App()
        {
            InitializeComponent();
        }

        protected override Window CreateWindow(IActivationState activationState)
        {
            return new Window(new AppShell());
        }
    }
```

As for your hanging references, I believe that's caused by SetterSpecificy. CC @StephaneDelcroix 

![image](https://github.com/dotnet/maui/assets/898335/e95c158b-56c3-4697-818e-860f873d618c)

**Comment by cgp1976** [comment]
yup indeed overriding the App.cs  CreateWindow  reflect the correct number of windows instance in the list.

ChildWindow. that's my bad. I should have name it ChildPage.

Some Findings I got during testing, basically I am just trying to find a working baseline.

replace the MainPage.xaml.cs Open_Window method with the following

 private void Open_Window(object sender, EventArgs e)
 {
     var page = new ChildWindow();

     Application.Current.OpenWindow(new Window(page));
   
     this.OnPropertyChanged(nameof(Count));
 }

with the above code. run the project again. click on the open window button first and from there onwards you can click on 
any buttons that suit you in every which way you like then click on invoke GC Collect button. 

 Go Diagnostic Tools > Memory Usage and 


Check for the ChildWindow - All Reference Instances  Count should match your context opened instance
Check for Microsoft.Maui.Controls.Window- All Reference Instances Count should match your context opened instance
